src/form/input/number/NumberInput.tsx

```diff
@@ -0,0 +1,157 @@
+  title?: string;
+  min?: number;
+  max?: number;
+  maxFractionDigits?: number;
+  placeholder?: string;
+  customClassName?: string;
+  leftIcon?: React.ReactNode;
+  onFocus?: InputProps["onFocus"];
+  onKeyDown?: InputProps["onKeyDown"];
+  prefixIconToValue?: string;
+  isDisabled?: boolean;
+  hasError?: boolean;
+}
+
+function NumberInput({
+  testid,
+  name,
+  title,
+  value,
+  min,
+  max,
+  maxFractionDigits = 0,
+  onChange,
+  hasError,
+  placeholder,
+  customClassName,
+  leftIcon,
+  onFocus,
+  onKeyDown,
+  prefixIconToValue,
+  isDisabled
+}: NumberInputProps) {
+  const className = classNames("input__number", customClassName);
+  const containerRef = useRef() as RefObject<HTMLDivElement>;
+
+  let finalValue = value;
+
+  if (finalValue && prefixIconToValue && !finalValue.includes(prefixIconToValue)) {
+    finalValue = `${prefixIconToValue} ${finalValue}`;
+  }
+
+  return (
+    <Input
+      inputContainerRef={containerRef}
+      testid={testid}
+      customClassName={className}
+      type={"text"}
+      title={title}
+      name={name}
+      onChange={handleChange}
+      value={finalValue || ""}
+      leftIcon={leftIcon}
+      placeholder={placeholder}
+      onFocus={handleFocus}
+      onBlur={handleBlur}
+      onKeyDown={handleKeyDown}
+      isDisabled={isDisabled}
+      hasError={hasError}
+      min={min}
+      max={max}
+    />
+  );
+
+  function handleBlur(event: React.SyntheticEvent<HTMLInputElement>) {
+    let {value: newValue} = event.currentTarget;
+
+    if (typeof max === "number" && parseFloat(value) > max) {
+      newValue = String(max);
+    }
+
+    if (typeof min === "number" && parseFloat(value) < min) {
+      newValue = String(min);
+    }
+
+    if (newValue !== event.currentTarget.value) {
+      event.currentTarget.value = newValue;
+      onChange(event);
+    }
+  }
+
+  function handleKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
+    if (onKeyDown) {
+      onKeyDown(event);
+    }
+
+    switch (event.key) {
+      case KEYBOARD_EVENT_KEY.ARROW_LEFT:
+      case KEYBOARD_EVENT_KEY.ARROW_RIGHT:
+      case KEYBOARD_EVENT_KEY.ARROW_UP:
+      case KEYBOARD_EVENT_KEY.ARROW_DOWN:
+        event.preventDefault();
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  function handleFocus(event: React.FocusEvent<HTMLInputElement>) {
+    if (onFocus) {
+      onFocus(event);
+    }
+  }
+
+  function handleChange(event: React.SyntheticEvent<HTMLInputElement>) {
+    const {value: newValue} = event.currentTarget;
+
+    let formattedNewValue = newValue
+      .replace(NOT_INTEGER_FIRST_CHARACTER_OF_STRING_REGEX, "")
+      .replace(NOT_NUMBER_NOR_DECIMAL_POINT_REGEX, "")
+      .replace(PRECISION_REGEX, "$1");
+
+    if (maxFractionDigits > 0) {
+      const decimalNumberParts = newValue.split(DECIMAL_NUMBER_SEPARATOR);
+      const decimalPart = decimalNumberParts[1];
+      const integerPart = decimalNumberParts[0].replace(THOUSANDTHS_SEPARATOR, "");
```

Comment:
These can stay for now but we will need to rethink these while we are working on the localization because `DECIMAL_NUMBER_SEPARATOR` and `THOUSANDTHS_SEPARATOR` changes for different locales